### ISSUE TITLE
Link checker fails on a matching substring

### ISSUE THREAD

**Issue by oleorhagen** [remarkjs/remark-validate-links#54]
### The checker accepts links with a matching substring

We've been doing some markdown migration (moving files around). Doing this I made a (pretty bad) script for helping me change up the paths, and fix the links automatically. At some point the script spit out this:
```
[foobar](../../../03.Devices/chapter.md/yocto-project/bootloader-support)
```
This was not caught by the internal link checker (this tool), and only by our dead-link checker on the site, once it was live.

I know it is a weird case, but I'd still consider this a bug.

### Your environment

*   **OS**: Ubuntu:18.04
*   **Packages**: `remark-validate-links, master`
*   **Env**: node v10.21.0

### Steps to reproduce

```test.md
[foobar](../thisdir/docs.md/foo/bar/baz)
```

`remark -u validate-links test.md

### Expected behaviour

Expected the tool to fail, as this is not a valid link.

Report it as a broken link as usual

### Actual behaviour

Passes due to the substring being valid (points to itself).

**Comment by wooorm** [comment]
Can you reduce the test case a bit more? From the path, I’m guessing that it has to do with `docs.md` being seen as a file, instead of a directory. Can you clarify which folders and files exist in that path?

**Comment by oleorhagen** [comment]
Is this better?

```
➜ scratchpad ls testfoo 
test.md
➜ scratchpad cat testfoo/test.md 


[foo](../testfoo/test.md/foo/bar/baz)
```

Then I run the checker on this file, and it does not fail.

**Comment by wooorm** [comment]
So this problem occurs when the current file is “entered” as a directory, with more directories and finally a file?

**Comment by oleorhagen** [comment]
> So this problem occurs when the current file is “entered” as a directory, with more non-existent things after?

Yes :)

**Comment by oleorhagen** [comment]
Also, it seems to have some problems with numbers in headers:
```
107:1-107:60  warning  Link to unknown heading: `step-1-create-a-deployment`. Did you mean `step-1---create-a-deployment`  missing-heading  remark-validate-links
```

But one thing at a time maybe?

**Comment by wooorm** [comment]
Thanks, should for the report. I believe it’s fixed in 10.0.1!

**Comment by oleorhagen** [comment]
Hi @wooorm Thank you so much for the fix :)

But could you add a prefix to the case also? I'll comment on the PR actually
